src/Windows/R/Components/Impl/Plots/Implementation/Commands/PlotHistoryCommand.cs

```diff
@@ -2,17 +2,50 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.Common.Core.Diagnostics;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.UI.Commands;
 using Microsoft.R.Components.InteractiveWorkflow;
 
 namespace Microsoft.R.Components.Plots.Implementation.Commands {
     internal abstract class PlotHistoryCommand : InteractiveWorkflowAsyncCommand {
         protected IRPlotHistoryVisualComponent VisualComponent { get; }
 
-        public PlotHistoryCommand(IRInteractiveWorkflow interactiveWorkflow, IRPlotHistoryVisualComponent visualComponent) :
+        protected PlotHistoryCommand(IRInteractiveWorkflow interactiveWorkflow, IRPlotHistoryVisualComponent visualComponent) :
             base(interactiveWorkflow) {
             Check.ArgumentNull(nameof(visualComponent), visualComponent);
             VisualComponent = visualComponent;
         }
+
+        public virtual CommandStatus Status {
+            get {
+                var selection = VisualComponent.SelectedPlots.ToArray();
+                if (selection.Length > 0 && selection.All(p => !p.ParentDevice.LocatorMode)) {
+                    return CommandStatus.SupportedAndEnabled;
+                }
+                return CommandStatus.Supported;
+            }
+        }
+
+        protected virtual bool CanInvoke() => true;
+
+        protected virtual Task InvokeAsync(IRPlot plot) => Task.CompletedTask;
+
+        public virtual async Task InvokeAsync() {
+            var selection = VisualComponent.SelectedPlots.ToArray();
+            if (selection.Length > 0 && CanInvoke()) {
+                try {
+                    foreach (var plot in selection) {
```

Comment:
All these 3 commands (`PlotHistoryActivateCommand`, `PlotHistoryRemoveCommand` and `PlotHistoryCutCopyCommand`) call our own functions in R: `rtvs:::graphics.ide.selectplot`, `rtvs:::graphics.ide.removeplot`, `rtvs:::graphics.ide.copyplot`. Instead of iterating through them inside command, I'd rather change signature of those functions so they can accept arrays.